acme/acme/client.py

```diff
@@ -224,7 +224,7 @@ def _revoke(self, cert, rsn, url):
                               messages.Revocation(
                                 certificate=cert,
                                 reason=rsn))
-        if response.status_code != http_client.OK:
+        if response.status_code != http_client.OK:  # pylint: disable=no-member
```

Comment:
I would not say Pylint is wrong on this specific error type. More that Pylint has several ways to understand that a member exists or not in a module, and the code fails to give to it any them.

So in my opinion, it is more an indication that the code may be designed better, and in that perspective, disabling entirely the check would be the solution to take if nothing more intellectually more satisfying can be done.

On that matter I have. There are similar 66 `no-name-in-module` errors due to `acme.magic_typing` making an `import *` on the `typing` module, while this module is not exposing a usable `__all__` dict. I wanted to fix that in a next PR to remove as much as possible `pylint: disable` directives, because Pylint is now smarter, upgrade dependencies help better Pylint, or simple redesign in our code can help better Pylint.

I propose to trade this 26 `disable=no-member` with the 66 `disable=no-name-in-module`.

If it really bother you to add all the `disable=no-member` directives, instead of disabling the error, I would largely prefer tweaking the `generated-members` option in Pylint that is designed for this situation.

Reply:
I'd love for the 66 incorrect `no-name-in-module` errors to be fixed too, but I think the 78 `no-member` errors is a separate issue. I don't want to trade fixing one bad behavior we already have for allowing a separate bad behavior to get worse in this PR with the new version of pylint.

I'm fine with taking the `generated-member` approach if you prefer, but I'm personally pretty put off by the idea of this check all together. I'm not sure it's even theoretically possible for pylint to get it right and the error here isn't even about our own objects.

Reply:
In fact it is a non-problem.

These `no-member` disable have been added in early stages of this PR, but I just discovered that most of them are in fact not needed. The few added ones are not enough in my opinion to remove this check.

In fact, from my tests, it seems that the majority of `pylint: disable=no-member` in certbot codebase can be removed.

Reply:
There are still 24 `disable=no-member` lines in our code and [many](https://github.com/PyCQA/pylint/issues/981) [upstream](https://github.com/PyCQA/pylint/issues/3134) [issues](https://github.com/PyCQA/pylint/issues/2224) [about](https://github.com/PyCQA/pylint/issues/2392) [false](https://github.com/PyCQA/pylint/issues/2425) [positives](https://github.com/PyCQA/pylint/issues/2425) (that's not all of them but that's enough to make a link out of every word in that phrase) so I'd still prefer to see the check disabled, but only two new disables are added in this PR so I'm not going to block it on that anymore.